AxonDeepSeg/data_management/input_data.py

```diff
@@ -1,437 +1,131 @@
-
-            # We load the image and discretize the masks
-            image, real_mask = self.prepare_image_mask()
+        return (image, mask)
 
-            # We apply data augmentation
-            augmented_data = copy.deepcopy(augmented_data_)
-            image, real_mask = self.apply_data_augmentation([image, real_mask], augmented_data, data_aug_verbose)
+    def __getitem__(self, index):
 
-            # Normalisation of the image
-            image = apply_legacy_preprocess(image)
-            #image = apply_preprocess(image, self.mean, self.variance)
-            # We save the obtained image and mask.
-            batch_x.append(image)
-            batch_y.append(real_mask)
-
-            # If we are at the end of an epoch, we reset the list of samples, so that during next epoch all sets will be different.
-            if self.sample_seen == self.epoch_size:
-                if each_sample_once:
-                    self.samples_list = self.reset_set(type_ = 'validation')
-                    break
-                else:
-                    self.samples_list = self.reset_set(type_ = 'train')
-
-        # Ensuring that we do have np.arrays of the good size for batch_x and batch_y before returning them
-        return transform_batches([batch_x, batch_y])
-
-
-    def next_batch_WithWeights(self, augmented_data_ = {'type':'None'},
-                               weights_modifier = {'balanced_activate':True, 'balanced_weights':[1.1, 1, 1.3],
-                                                   'boundaries_activate':False},
-                               each_sample_once=False, data_aug_verbose=0):
-
-        """
-        :param weights_modifier:
-        :param augmented_data: if True, each patch of the batch is randomly transformed with the data augmentation process.
-        :return: The triplet [batch_x (data), batch_y (prediction), weights (based on distance to edges)] to feed the network.
         """
-        batch_x = []
-        batch_y = []
-        batch_w = []
-
-        for i in range(self.batch_size) :
-
-            # We prepare the image and the corresponding mask by discretizing the mask.
-            image, real_mask = self.prepare_image_mask()
-
-            # Generation of the weights map
-            real_weights = self.generate_weights_map(weights_modifier, real_mask)
-
-            # Application of data augmentation
-            augmented_data = copy.deepcopy(augmented_data_)
-
-            image, real_mask, real_weights = self.apply_data_augmentation([image, real_mask, real_weights],
-                                                                          augmented_data, data_aug_verbose)
+        data_gen_args = dict(vertical_flip=True)
+        image_datagen = ImageDataGenerator(**data_gen_args)
+        mask_datagen = ImageDataGenerator(**data_gen_args)
 
-            # Normalisation of the image
-            image = apply_legacy_preprocess(image)
+        seed = 2018
```

Comment:
Any reason why using this hardcoded seed value?

Reply:
When I was using jupyter notebook to produce consistent predictions, I was using seed. But I don't think here its necessary. I will fix this.